### ISSUE TITLE
Exception when serializing property of type object using JObject.Parse.

### ISSUE THREAD

**Issue by santoshkaranam** [Azure/autorest#86]
I have a web API whose return type is DTO1. DTO1 has a property P1 of type object

I have auto generated webapi client using AutoRest.0.9.7

When i try to make a request to webAPi using auto generated client, it gives exception.

Exception when serializing property P1 using JObject.Parse.

The Value of P1 at point when exception was thrown was "value1"...

I believe there is not way to convert string value ("value1") using JObject.Parse.

JObject.Parse always expects the string passed to be a json string.

On contrary when i use JsonConter.Serialize , the serialization works fine..

**Comment by anobleperson** [comment]
I have a similar issue. and I haven't figured out the magic sauce to use a JSON string yet either.  The following test code fails on the Serialize line with the error:
> "Newtonsoft.Json.JsonReaderException : Invalid JavaScript property identifier character: }. Path '', line 1, position 2."

```C#
        [Test]
        public void TestObjectSerialization()
        {
            var item = new TestClass{MyObjectValue = "{1}"};
            var serialized = item.SerializeJson(null);
            item.DeserializeJson(serialized);

            Assert.AreEqual(1, JsonConvert.DeserializeObject(item.MyObjectValue, typeof(object)));
        }
```
The example controller is:
```C#
    public class ObjectTestingController : ApiController
    {
        [HttpPost]
        [Route("api/TestingObjects")]
        public TestClass TestingObjects(TestClass input)
        {
            return input;
        }

        public class TestClass
        {
            public object MyObjectValue { get; set; }
        }
    }
```
Which has the following swagger:
```JSON
		"/api/TestingObjects" : {
			"post" : {
				"tags" : ["ObjectTesting"],
				"operationId" : "ObjectTesting_TestingObjects",
				"consumes" : ["application/json", "text/json", "application/xml", "text/xml", "application/x-www-form-urlencoded"],
				"produces" : ["application/json", "text/json", "application/xml", "text/xml"],
				"parameters" : [{
						"name" : "input",
						"in" : "body",
						"required" : true,
						"schema" : {
							"$ref" : "#/definitions/TestClass"
						}
					}
				],
				"responses" : {
					"200" : {
						"description" : "OK",
						"schema" : {
							"$ref" : "#/definitions/TestClass"
						}
					}
				},
				"deprecated" : false
			}
		},
...
		"TestClass" : {
			"type" : "object",
			"properties" : {
				"MyObjectValue" : {
					"$ref" : "#/definitions/Object"
				}
			}
		},
```
And the generated serialize code is:
```C#
    public partial class TestClass
    {
        private string _myObjectValue;
        
        /// <summary>
        /// Optional.
        /// </summary>
        public string MyObjectValue
        {
            get { return this._myObjectValue; }
            set { this._myObjectValue = value; }
        }
        
        /// <summary>
        /// Initializes a new instance of the TestClass class.
        /// </summary>
        public TestClass()
        {
        }
        
        /// <summary>
        /// Serialize the object
        /// </summary>
        /// <returns>
        /// Returns the json model for the type TestClass
        /// </returns>
        public virtual JToken SerializeJson(JToken outputObject)
        {
            if (outputObject == null)
            {
                outputObject = new JObject();
            }
            if (this.MyObjectValue != null)
            {
                outputObject["MyObjectValue"] = JObject.Parse(this.MyObjectValue);
            }
            return outputObject;
        }
        
        /// <summary>
        /// Deserialize the object
        /// </summary>
        public virtual void DeserializeJson(JToken inputObject)
        {
            if (inputObject != null && inputObject.Type != JTokenType.Null)
            {
                JToken myObjectValueValue = inputObject["MyObjectValue"];
                if (myObjectValueValue != null && myObjectValueValue.Type != JTokenType.Null)
                {
                    this.MyObjectValue = myObjectValueValue.ToString(Newtonsoft.Json.Formatting.Indented);
                }
            }
        }
    }
```

It would be preferable to be able to use object types, so a possible solution is to generate a property with the following serialize/deserialize:
```C#
   public partial class TestClass
    {
        private object _myObjectValue;
        
        /// <summary>
        /// Optional.
        /// </summary>
        public object MyObjectValue
        {
            get { return this._myObjectValue; }
            set { this._myObjectValue = value; }
        }
        
        /// <summary>
        /// Initializes a new instance of the TestClass class.
        /// </summary>
        public TestClass()
        {
        }
        
        /// <summary>
        /// Serialize the object
        /// </summary>
        /// <returns>
        /// Returns the json model for the type TestClass
        /// </returns>
        public virtual JToken SerializeJson(JToken outputObject)
        {
            if (outputObject == null)
            {
                outputObject = new JObject();
            }
            if (this.MyObjectValue != null)
            {
                outputObject["MyObjectValue"] = new JValue(this.MyObjectValue);
            }
            return outputObject;
        }
        
        /// <summary>
        /// Deserialize the object
        /// </summary>
        public virtual void DeserializeJson(JToken inputObject)
        {
            if (inputObject != null && inputObject.Type != JTokenType.Null)
            {
                JToken myObjectValueValue = inputObject["MyObjectValue"];
                if (myObjectValueValue != null && myObjectValueValue.Type != JTokenType.Null)
                {
                    this.MyObjectValue = JsonConvert.DeserializeObject(myObjectValueValue.ToString(), typeof(object));
                }
            }
        }
    }
```

**Comment by hovsepm** [comment]
Serialization and deserialization has been completely redesigned to use Newtonsoft.Json SerializeObject and DeserializeObject. With the upcoming release of AutoRest nested properties of object type will be supported.

**Comment by hovsepm** [comment]
Fixed in Release 0.10.0 - [nuget](https://www.nuget.org/packages/AutoRest/), [chocolatey](https://chocolatey.org/packages/AutoRest), [github] (https://github.com/Azure/autorest/releases/tag/AutoRest-0.10.0)
